fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -610,8 +610,8 @@ public Response read(@PathParam("type") String type, @PathParam("id") String id)
 
         try {
             checkInitComplete();
-
-            Resource resource = doRead(type, id, true, false, null, null);
+            MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
```

Comment:
This should be available in the FHIRPersistenceContext.get().getFHIRPersistenceEvent(PROPNAME_REQUEST_PROPERTIES)

I think this is probably the preferred way, rather than add to the signature method.

Reply:
not really sure what's your meaning... 
According to the function comments of doSearch, seems requestProperties is used for the http headers only, not sure how it can be used for the query parameters... 

 * @param queryParameters
     *            a Map containing the query parameters from the request URL
     * @param requestProperties
     *            additional request properties which supplement the HTTP headers associated with this request

Reply:
I'm suggesting that there is an existing parameter/property. Or should be extended to add the property to the context, rather than put in the method.

Reply:
Discussed with Albert, no problem with code, just a pattern conversation, resolved to leave as implemented in PR